jdaviz/components/layer_viewer_icon_stylized.vue

```diff
@@ -0,0 +1,106 @@
+<template>
+  <j-tooltip :tooltipcontent="tooltipContent(tooltip, label, visible, colormode, colors, linewidth, is_subset)">
+    <v-btn
+      :rounded="is_subset"
+      @click="(e) => $emit('click', e)"
+      :style="'padding: 0px; margin-bottom: 4px; background: '+visibilityBackgroundStyle(visible)+', '+colorBackgroundStyle(colors, cmap_samples)+'; '+btn_style"
+      width="30px"
+      min-width="30px"
+      height="30px"
+    >
+      <span :style="'color: white; text-shadow: 0px 0px 3px black; '+borderStyle(linewidth)">
+        {{ icon }}
+      </span>
+    </v-btn>
+  </j-tooltip>
+</template>
+
+<script>
+module.exports = {
+  // tooltip: undefined will use default generated, empty will skip tooltips, any other string will be used directly
+  props: ['label', 'icon', 'visible', 'is_subset', 'colors', 'linewidth', 'colormode', 'cmap_samples', 'btn_style', 'tooltip'],
+  methods: {
+    tooltipContent(tooltip, label, visible, colormode, colors, linewidth, is_subset) {
+      if (tooltip !== undefined) {
+        return tooltip
+      }
+      var tooltip = label
+      if (visible === 'mixed') {
+        tooltip += '<br/>Visibility: mixed'
+      } else if (!visible) {
+        tooltip += '<br/>Visibility: hidden'
+      }
+      if (colormode === 'mixed' && !is_subset) {
+        tooltip += '<br/>Color mode: mixed'
+      }
+      if (colors.length > 1) {
+        if (colormode === 'Colormaps' && !is_subset) {
+          tooltip += '<br/>Colormap: mixed'
+        } else if (colormode === 'mixed' && !is_subset) {
+          tooltip += '<br/>Color/colormap: mixed'
+        } else {
+          tooltip += '<br/>Color: mixed'
+        }
+      }
+      if (linewidth == 'mixed' && !is_subset) {
+        tooltip += '<br/>Linewidth: mixed'
+      }
+      return tooltip
+    },
+    visibilityBackgroundStyle(visible) {
+      if (visible === 'mixed'){
+        return 'repeating-linear-gradient(30deg, rgba(0,0,0,0.3), rgba(0,0,0,0.3) 3px, transparent 3px, transparent 3px, transparent 10px)'
+      }
+      else if (visible) {
+        return 'repeating-linear-gradient(30deg, transparent, transparent 10px)'
+      } else {
+        return 'repeating-linear-gradient(30deg, rgba(0,0,0,0.4), rgba(0,0,0,0.4) 8px, transparent 8px, transparent 8px, transparent 10px)'
+      }
+    },
+    colorBackgroundStyle(colors, cmap_samples) {
```

Comment:
I don't expect that these loops to evaluate the colormap for the icons cost much. But is there a way to prove to ourselves that this is true?

Reply:
we could try to profile it?  All this is doing though is taking the input lists generated in python and joining them into a string, with a bit of math to set the widths of each entry, so I really don't expect it to show up in anything.